src/main/java/org/generationcp/middleware/dao/ims/TransactionDAO.java

```diff
@@ -741,8 +743,24 @@ private String formatDate(final Date date) {
 		try {
 			return TransactionDto.class.getConstructor(Integer.class, String.class, String.class,  Double.class,
 			 String.class, Date.class, Integer.class, Integer.class, String.class, String.class, Integer.class, String.class, String.class);
-		} catch (NoSuchMethodException ex) {
+		} catch (final NoSuchMethodException ex) {
 			throw new RuntimeException(ex);
 		}
 	}
+
+
+	public Transaction saveTransaction(final TransactionDto transactionDto, final Lot lot, final TransactionStatus transactionStatus) {
+		final Transaction transaction = new Transaction();
+		transaction.setStatus(transactionStatus.getIntValue());
+		transaction.setLot(lot);
+		transaction.setPersonId(Integer.valueOf(transactionDto.getUser()));
+		transaction.setUserId(Integer.valueOf(transactionDto.getUser()));
+		transaction.setTransactionDate(new Date());
+		transaction.setQuantity(transactionDto.getAmount());
+		transaction.setPreviousAmount(0D);
+		transaction.setCommitmentDate(0);
+		transaction.setComments(transactionDto.getNotes());
```

Comment:
@vmaletta not so sure about commitment date always being 0, I guess it depends on the TransactionType